quickwit/quickwit-search/src/leaf.rs

```diff
@@ -446,88 +539,151 @@ pub async fn leaf_search(
     searcher_context: Arc<SearcherContext>,
     request: Arc<SearchRequest>,
     index_storage: Arc<dyn Storage>,
-    splits: &[SplitIdAndFooterOffsets],
+    splits: &mut [SplitIdAndFooterOffsets],
```

Comment:
I understand that you need it to be mut to sort it.
Let's pass ownership. The `&mut []` makes the API confusing.


```suggestion
    mut splits: Vec<SplitIdAndFooterOffsets>,
```